### PR TITLE
Idle timeout

### PR DESCRIPTION
**Pull request by spirali**

### CODE REVIEW

File: src/server/rpc.rs [hunk 1]
```diff
@@ -129,28 +129,45 @@ async fn worker_rpc_loop<
         let mut interval = tokio::time::interval(heartbeat_interval);
```
Comment (Kobzol) [inline src/server/rpc.rs:129]: Maybe this should be
```rust
let mut interval = tokio::time::interval(std::cmp::min(heartbeat_interval, core.idle_timeout().unwrap_or(heartbeat_interval)));
```
?

File: src/messages/gateway.rs [hunk 2]
```diff
@@ -161,6 +161,7 @@ pub struct NewWorkerMessage {
 pub struct LostWorkerMessage {
     pub worker_id: WorkerId,
     pub running_tasks: Vec<TaskId>,
+    pub message: String, // Details about reason why worker is lost
```
Comment (Kobzol) [inline src/messages/gateway.rs:164]: It would maybe be better if this was an enum (e.g. for nicer worker error state formatting in HQ tables).

**Review by Kobzol — CHANGES_REQUESTED** [review]

File: src/server/rpc.rs (hunk 1 above)
Comment (spirali) [reply src/server/rpc.rs:129]: You are right. I had some internals reasons why not not do it in this way, but when I was writing them here, I realized that it is a bullshit :) [I generally plan to make this interval dynamically configurable, which makes the interval slightly tricky, but this is a problem for our future selves]

File: src/messages/gateway.rs (hunk 2 above)
Comment (spirali) [reply src/messages/gateway.rs:164]: Good point

File: src/server/worker.rs [hunk 3]
```diff
@@ -16,6 +16,7 @@ bitflags::bitflags! {
         // There is no "waiting" task for resources of this worker
         // Therefore this worker should not cause balancing even it is underloaded
         const PARKED = 0b00000001;
+        const STOPPING = 0b00000010;
```
Comment (Kobzol) [inline src/server/worker.rs:19]: Maybe add a comment here? `STOPPING` sounds like it's in the process of being stopped, but it's unclear why and what caused it.

**Pull request merged**